contrib/pdf_slideshow.lua

```diff
@@ -0,0 +1,229 @@
+  if not res then
+    error(errmsg)
+  end
+end
+
+local function slide(latexfile,i,image,file)
+  local title = ""
+  if title_widget.text ~= "" then
+    title = title_widget.text
+  end
+  if check_button_author.value == true and image.creator ~= "" then
+    if title ~= "" then
+       title = title.." / "
+    end
+    title = title..image.creator
+  end
+  if check_button_title.value == true and image.title ~= "" then
+    if title ~= "" then
+       title = title.." / "
+    end
+    title = title..image.title
+  end
+
+  --  make sure there is no non-printable characters in the title
+  title = string.gsub(title, "\n", "")
+  title = string.gsub(title, "\r", "")
+  title = string.gsub(title, "\t", "")
+
+  --  fact is that latex will get confused if the filename has multiple dots.
+  --  so \includegraphics{file.01.jpg} wont work. We need to output the filename
+  --  and extention separated, e.g: \includegraphics{{file.01}.jpg}
+  local filenoext=string.gsub(file, "(.*)(%..*)", "%1")
+  local ext=string.gsub(file, "(.*)(%..*)", "%2")
+  my_write(latexfile,"\\begin{minipage}[b]{0,99\\textwidth}\n")
+
+  my_write(latexfile,"\\includegraphics[height=0.95\\textheight]{{"..filenoext.."}"..ext.."}\\newline\n")
+  my_write(latexfile,"\\centering{{\\color{white}\\verb|"..title.."|}}\n")
+  my_write(latexfile,"\\end{minipage}\\quad\n")
+end
+
+dt.register_storage("pdf_slideshow",_("pdf slideshow"),
+    nil,
+    function(storage,image_table)
+
+    local preamble = [[
+    \documentclass[a4paper,10pt,landscape]{beamer}
+    \usetheme{default}
+    \usepackage{graphicx}
+    \usepackage[space]{grffile} % needed to support filename with spaces
+    \pagestyle{empty}
+    \parindent0pt
+    \usepackage{geometry}
+    \usepackage{color}
+    \pagecolor{black!100}
+    \color{white}
+    \geometry{a4paper,landscape,left=5mm,right=5mm, top=5mm, bottom=5mm}
+    \mode<presentation>
+    \transduration{]]..delay_widget.value..[[}
+    \setbeamertemplate{footline}{}
+    \setbeamertemplate{headline}{}
+    \setbeamertemplate{frametitle}{}
+    \setbeamertemplate{navigation symbols}{}
+    \setbeamercolor{background canvas}{fg=white,bg=black!100}
+    \setbeamercolor{normal text}{fg=white}
+    \hypersetup{pdfstartpage=1,pdfpagemode=FullScreen}
+    \begin{document}
+    ]]
+
+    local errmsg
+    local latexfile
+    latexfile,errmsg=io.open(filename,"w")
+    if not latexfile then
+        error(errmsg)
+      end
+      my_write(latexfile,preamble)
+      local i = 1
+      for img,file in pairs(image_table) do
+         slide(latexfile,i,img,file)
+         my_write(latexfile,"\n\\bigskip\n")
+         i = i+1
+      end
+      my_write(latexfile,ending)
+      latexfile:close()
+
+      -- convert to PDF
+      local dir=string.gsub(filename, "(.*/)(.*)", "%1")
+      local locfile=string.gsub(filename, "(.*/)(.*)", "%2")
+      local command = "pdflatex -halt-on-error -output-directory "..dir.." "..locfile
```

Comment:
You should check if pdflatex is installed. You can use the function from dtutils.file:
Just add "local df = require "lib/dtutils.file"" at the top and then do a check like this:
  if not df.check_if_bin_exists("pdflatex") then
    dt.print_error(_("pdflatex not found"))
    return
end	

Here you can see an example:
https://github.com/darktable-org/lua-scripts/blob/master/contrib/geoToolbox.lua#L313